components/address-input.jsx

```diff
@@ -624,12 +624,15 @@ var Country = React.createClass({
     this.setState({
       valid: true
     });
+    // Changing the country should clear any errors on province.
+    form.error("province", "");
```

Comment:
If province was an error, changing country should clear any errors on the country dropdown left over from the previous country's list of provinces, if any.